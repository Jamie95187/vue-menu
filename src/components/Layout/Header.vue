<template>
  <div>
    <v-navigation-drawer
      v-model="sideNav"
      absolute
      temporary
      >
      <v-list>
        <v-list-item
          v-for="item in menuItems"
          :key="item.title"
          :to="item.link">
          <v-list-item-action>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-item-action>
          <v-list-item-content>{{ item.title }}</v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>
      <v-app-bar app class="primary">
        <router-link to="/" tag="span" style="cursor: pointer">
          <v-toolbar-title class="white--text">
            Sushi Menu
          </v-toolbar-title>
        </router-link>
        <v-app-bar-nav-icon
          @click.stop="sideNav = !sideNav"
          ></v-app-bar-nav-icon>
        <v-spacer></v-spacer>
        <v-btn
          text class="white--text"
          v-for="item in menuItems"
          :key="item.title"
          :to="item.link">
          <v-icon left>
            {{ item.icon }}
          </v-icon>
          {{ item.title }}
        </v-btn>
      </v-app-bar>
    </div>
</template>

<script>
  export default {
    name: "Header",
    data: () => ({
      sideNav: false,
    }),
    // Show different page for authenticated vs unauthenticated users
    computed: {
      menuItems() {
        // default case when not logged in
        let menuItems = [
          { icon: 'face', title: 'Sign up', link: '/signup'},
          { icon: 'lock_open', title: 'Sign in', link: '/signin'},
          { icon: 'fastfood', title: 'Menu', link: '/'},
          { icon: 'description', title: 'Order', link: '/order'}
        ]
        if (this.userIsAuthenticated) {
          menuItems = [
            { icon: 'supervisor_account', title: 'View Order', link: '/order'},
            { icon: 'person', title: 'Profile', link: '/profile'},
            { icon: 'fastfood', title: 'Menu', link: '/'},
            { icon: 'description', title: 'Order', link: '/order'}
          ]
        }
        return menuItems
      },
      userIsAuthenticated () {
        return this.$store.getters.user !== null && this.$store.getters.user !== undefined
      }
    }
  }

</script>
