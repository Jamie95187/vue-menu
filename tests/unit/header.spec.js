import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Layout/Header.vue'
import Vue from 'vue'
import Vuetify from 'vuetify';

Vue.use(Vuetify);

const localVue = createLocalVue()

describe('Header.vue', () => {
  let wrapper;

  it('renders Header Component', () => {
    wrapper = shallowMount(Header);
    expect(wrapper.exists()).toBe(true)
  })

})
