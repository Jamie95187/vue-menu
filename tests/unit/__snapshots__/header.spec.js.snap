// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue should match match the snapshot 1`] = `
<v-layout-stub>
  <v-navigation-drawer-stub absolute="" temporary="">
    <v-list-stub>
      <v-list-item-stub to="/signup">
        <v-list-item-action-stub>
          <v-icon-stub>face</v-icon-stub>
        </v-list-item-action-stub>
        <v-list-item-content-stub>Sign up</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub to="/signin">
        <v-list-item-action-stub>
          <v-icon-stub>lock_open</v-icon-stub>
        </v-list-item-action-stub>
        <v-list-item-content-stub>Sign in</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub to="/">
        <v-list-item-action-stub>
          <v-icon-stub>fastfood</v-icon-stub>
        </v-list-item-action-stub>
        <v-list-item-content-stub>Menu</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub to="/order">
        <v-list-item-action-stub>
          <v-icon-stub>mdi-basket</v-icon-stub>
        </v-list-item-action-stub>
        <v-list-item-content-stub>View Order</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub to="/orders">
        <v-list-item-action-stub>
          <v-icon-stub>mdi-history</v-icon-stub>
        </v-list-item-action-stub>
        <v-list-item-content-stub>Order History</v-list-item-content-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-navigation-drawer-stub>
  <v-app-bar-stub app="" class="primary">
    <router-link-stub to="/" tag="span" style="cursor: pointer;">
      <v-toolbar-title-stub class="white--text">
        Sushi Menu
      </v-toolbar-title-stub>
    </router-link-stub>
    <v-app-bar-nav-icon-stub></v-app-bar-nav-icon-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub text="" to="/signup" class="white--text">
      <v-icon-stub left="">
        face
      </v-icon-stub>
      Sign up
    </v-btn-stub>
    <v-btn-stub text="" to="/signin" class="white--text">
      <v-icon-stub left="">
        lock_open
      </v-icon-stub>
      Sign in
    </v-btn-stub>
    <v-btn-stub text="" to="/" class="white--text">
      <v-icon-stub left="">
        fastfood
      </v-icon-stub>
      Menu
    </v-btn-stub>
    <v-btn-stub text="" to="/order" class="white--text">
      <v-icon-stub left="">
        mdi-basket
      </v-icon-stub>
      View Order
    </v-btn-stub>
    <v-btn-stub text="" to="/orders" class="white--text">
      <v-icon-stub left="">
        mdi-history
      </v-icon-stub>
      Order History
    </v-btn-stub>
    <!---->
  </v-app-bar-stub>
</v-layout-stub>
`;
